packages/core/upload/admin/src/components/MediaLibraryInput/index.jsx

```diff
@@ -179,38 +179,44 @@ export const MediaLibraryInput = forwardRef(
           trackedLocation="content-manager"
         />
 
-        <AssetDialog
-          allowedTypes={fieldAllowedTypes}
-          initiallySelectedAssets={initiallySelectedAssets}
-          folderId={folderId}
-          onClose={() => {
-            setStep(undefined);
-            setFolderId(null);
-          }}
-          open={step === STEPS.AssetSelect}
-          onValidate={handleValidation}
-          multiple={multiple}
-          onAddAsset={() => setStep(STEPS.AssetUpload)}
-          onAddFolder={() => setStep(STEPS.FolderCreate)}
-          onChangeFolder={(folder) => setFolderId(folder)}
-          trackedLocation="content-manager"
-        />
-
-        <UploadAssetDialog
-          open={step === STEPS.AssetUpload}
-          onClose={() => setStep(STEPS.AssetSelect)}
-          initialAssetsToAdd={droppedAssets}
-          addUploadedFiles={handleFilesUploadSucceeded}
-          trackedLocation="content-manager"
-          folderId={folderId}
-          validateAssetsTypes={validateAssetsTypes}
-        />
-
-        <EditFolderDialog
-          open={step === STEPS.FolderCreate}
-          onClose={() => setStep(STEPS.AssetSelect)}
-          parentFolderId={folderId}
-        />
+        {step === STEPS.AssetSelect && (
+          <AssetDialog
+            allowedTypes={fieldAllowedTypes}
+            initiallySelectedAssets={initiallySelectedAssets}
+            folderId={folderId}
+            onClose={() => {
+              setStep(undefined);
+              setFolderId(null);
+            }}
+            open={step === STEPS.AssetSelect}
```

Comment:
no that's the exact pbl, inside the component all the states are preserved even if we close because it never unmounts. the long term improvement is to have a clearState inside the compo to be able to just use the open prop but not the right timing for this